eth_utils/currency.py

```diff
@@ -61,7 +61,7 @@ def from_wei(number: int, unit: str) -> Union[int, decimal.Decimal]:
     return result_value
 
 
-def to_wei(number: int, unit: str) -> int:
+def to_wei(number: int, unit: str) -> Union[int, float, str, decimal.Decimal]:
```

Comment:
Hey @Christofon, thanks for the pr. I think you're close, but it's only the input arg `number` which was too strict and needs updating, since an `int` is always returned on line 85 / 102. 

It should look something like this
> def to_wei(number: Union[int, float, str, decimal.Decimal], unit: str) -> int:

Reply:
thanks for the explanation, just corrected it.